### PR TITLE
CSVExporter

### PR DESCRIPTION
**Pull request by roryhofland**
# Description

Worker class to export a schedule in a CSV file format.

## Type of change

Please delete options that are not relevant.

- [x] New feature 
- [x] This change requires a documentation update

# How Has This Been Tested?

Please cite/describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] CSVExporterTest. Just make a dummy CSV file and compare it to one that is generated.

# Does this PR follow the rubric?

Yes

Please describe any SOLID principle, design pattern, etc. used in the PR that will be added to the design document. If it does, **please link in design doc!!!**

Template method (already linked)

# Checklist:

- [x] I ran `./gradlew spotlessApply`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I ran `./gradlew build` and the gradle build succeeded
- [x] I have not violated any design principles

# Other Comments:

Other comments that should be made clear.

### CODE REVIEW

File: src/main/java/workers/CSVExporter.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import util.InvalidSessionException;
+
+public class CSVExporter extends Exporter {
+
+    private int num = 0;
```
Comment (roryhofland) [inline src/main/java/workers/CSVExporter.java:15]: Could be added to the Exporter superclass.

File: src/main/java/workers/CSVExporter.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    private final ZoneId zoneId = ZoneId.of("-4");
+    private final LocalDate now = LocalDate.now(zoneId);
+    private final int startYear = now.getMonthValue() < 9 ? now.getYear() - 1 : now.getYear();
+    private final int endYear = startYear + 1;
+    private final LocalDate FALL_SEMESTER_START_DATE = LocalDate.of(startYear, 9, 9);
+    private final LocalDate FALL_SEMESTER_END_DATE = LocalDate.of(startYear, 12, 10);
+    private final LocalDate WINTER_SEMESTER_START_DATE = LocalDate.of(endYear, 1, 10);
+    private final LocalDate WINTER_SEMESTER_END_DATE = LocalDate.of(endYear, 4, 11);
```
Comment (roryhofland) [inline src/main/java/workers/CSVExporter.java:23]: These could be added to superclass.

Comment (kenneth-miura) [reply src/main/java/workers/CSVExporter.java:23]: yeah you're right

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
Sorry for late review, was getting rocked by csc258. Looks good in general, I have one point about the arguments for `outputSchedule`.

File: src/main/java/controllers/CommandLineInterface.java [hunk 3]
```diff
@@ -225,6 +227,11 @@ public static void displayUserSchedules(List<Schedule> userSchedules) {
                     System.out.println("Saving this schedule in .ics format...");
                     ScheduleExporter.outputScheduleICS(currSchedule);
                     break;
+                case 'C':
```
[the remark this answers is not in the document]
Comment (roryhofland) [reply src/main/java/controllers/CommandLineInterface.java:230]: yeah pretty much. for actual calendar stuff .ics is better.

**Comment by sugolov** [comment]
Looks good to me, very nice

**Pull request merged**